graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/tweetfeature/TopSecondDegreeByCountForTweet.java

```diff
@@ -0,0 +1,142 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.twitter.graphjet.algorithms.counting.tweetfeature;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.twitter.graphjet.algorithms.NodeInfo;
+import com.twitter.graphjet.algorithms.RecommendationInfo;
+import com.twitter.graphjet.algorithms.RecommendationType;
+import com.twitter.graphjet.algorithms.counting.TopSecondDegreeByCount;
+import com.twitter.graphjet.algorithms.counting.TopSecondDegreeByCountResponse;
+import com.twitter.graphjet.algorithms.counting.tweet.TopSecondDegreeByCountRequestForTweet;
+import com.twitter.graphjet.algorithms.filters.RecentTweetFilter;
+import com.twitter.graphjet.bipartite.NodeMetadataMultiSegmentIterator;
+import com.twitter.graphjet.bipartite.RightNodeMetadataLeftIndexedMultiSegmentBipartiteGraph;
+import com.twitter.graphjet.bipartite.api.EdgeIterator;
+import com.twitter.graphjet.hashing.IntArrayIterator;
+import com.twitter.graphjet.stats.StatsReceiver;
+
+public class TopSecondDegreeByCountForTweet extends
+  TopSecondDegreeByCount<TopSecondDegreeByCountRequestForTweet, TopSecondDegreeByCountResponse> {
+  // Max number of node metadata associated with each right node.
+  private static final int MAX_NUM_METADATA = 200;
+
+  /**
+   * Initialize all the states needed to run TopSecondDegreeByCountForTweet. Note that the object can
+   * be reused for answering many different queries on the same graph, which allows for
+   * optimizations such as reusing internally allocated maps etc.
+   *
+   * @param leftIndexedBipartiteGraph is the
+   *                                  {@link RightNodeMetadataLeftIndexedMultiSegmentBipartiteGraph}
+   *                                  to run TopSecondDegreeByCountForTweet on
+   * @param expectedNodesToHit        is an estimate of how many nodes can be hit in
+   *                                  TopSecondDegreeByCountForTweet. This is purely for allocating needed
+   *                                  memory right up front to make requests fast.
+   * @param statsReceiver             tracks the internal stats
+   */
+  public TopSecondDegreeByCountForTweet(
+    RightNodeMetadataLeftIndexedMultiSegmentBipartiteGraph leftIndexedBipartiteGraph,
+    int expectedNodesToHit,
+    StatsReceiver statsReceiver
+  ) {
+    super(leftIndexedBipartiteGraph, expectedNodesToHit, statsReceiver);
+  }
+
+  @Override
+  protected boolean isEdgeUpdateValid(
+    TopSecondDegreeByCountRequestForTweet request,
+    long rightNode,
+    byte edgeType,
+    long edgeMetadata
+    ) {
+    long timeStampFromTweetId = RecentTweetFilter.originalTimeStampFromTweetId(rightNode);
+    // if the timestamp of the right node exceeds the right node age limit;
+    if (timeStampFromTweetId < (System.currentTimeMillis() - request.getMaxRightNodeAgeInMillis())) {
```

Comment:
This isn't necessarily an edge filter, but a node filter instead. If there are new engagements to an old tweet, it would be filtered out because the tweet itself is too old. I believe you are either looking for RecentTweetFilter through filterChain, or this is not what you intended?